csharp/src/Apache.Arrow/Ipc/ArrowFileWriter.cs

```diff
@@ -67,10 +70,27 @@ public override async Task WriteRecordBatchAsync(RecordBatch recordBatch, Cancel
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var block = await WriteRecordBatchInternalAsync(recordBatch, cancellationToken)
+            await WriteRecordBatchInternalAsync(recordBatch, cancellationToken)
                 .ConfigureAwait(false);
+        }
+
+        private protected override void StartingWritingRecordBatch()
+        {
+            _currentRecordBatchOffset = BaseStream.Position;
+        }
+
+        private protected override void FinishedWritingRecordBatch(long bodyLength, long metadataLength)
+        {
+            Debug.Assert(_currentRecordBatchOffset > 0);
```

Comment:
It can't be == 0?

Reply:
It can't because there is always a schema header that needs to get written before any Record Batch can be written.

Do you think I should add this to a comment?

Reply:
Thanks.  A comment or message in the assert seems useful.